app/app/search.py

```diff
@@ -154,38 +161,40 @@ def find_in_predicted(
 
 @cache
 def get_chemical_classes() -> list[Tuple[str, str]]:
-    all_compounds = select(ValidatedCompounds.compound_id).distinct().subquery()
+    all_compounds = select(ValidatedCompounds.compound_id).distinct()
     stmt = select(Compounds.chemical_class).filter(Compounds.compound_id.in_(all_compounds)).distinct()
     with db_session() as session:
         chemical_classes = session.execute(stmt)
         return [
-            (f"class: {cc}", f"class:{cc}")
+            (f"class: {cc}", cc)
             for cc, in chemical_classes
         ]
 
 
 @cache
-def get_compounds() -> list[Tuple[str, str]]:
-    all_compounds = select(ValidatedCompounds.compound_id).distinct().subquery()
-    stmt = select(Compounds.compound_name).filter(Compounds.compound_id.in_(all_compounds)).distinct()
+def get_compounds() -> list[Tuple[str, str, str]]:
+    all_compounds = select(ValidatedCompounds.compound_id).distinct()
+    stmt = select(Compounds.compound_name, Compounds.chemical_class, Compounds.cas_number).filter(Compounds.compound_id.in_(all_compounds)).distinct()
```

Comment:
Sorting helps the user find the options:
```suggestion
    stmt = select(Compounds.compound_name, Compounds.chemical_class, Compounds.cas_number).filter(Compounds.compound_id.in_(all_compounds)).distinct().order_by(collate(Compounds.compound_name, 'NOCASE'))
```

Reply:
This should be resolved.